src/app/story/directives/status-control.component.ts

```diff
@@ -0,0 +1,356 @@
+          // promoting from Draft to Scheduled
+          this.buttons = {
+            primary: {
+              text: 'Schedule',
+              action: () => this.saveAndPublish('scheduled for publishing'),
+              disabled: false,
+              color: 'green'
+            }
+          }
+        } else {
+          // already scheduled
+          this.buttons = {
+            primary: {
+              text: 'Save',
+              action: () => this.save(),
+              disabled: !this.isChanged,
+              color: 'green'
+            }
+          }
+        }
+      }
+      break;
+    //////////// PUBLISHED ////////////
+    case 'published':
+      if (this.strictInvalid) {
+        this.buttons = {
+          primary: {
+            text: this.countProblems(true),
+            action: () => this.showProblems(),
+            disabled: false,
+            color: 'red'
+          }
+        }
+      } else {
+        if (this.currentStatus !== 'published') {
+          this.buttons = {
+            primary: {
+              text: 'Publish Now',
+              action: () => this.saveAndPublish('published'),
+              disabled: false,
+              // color: 'blue' // button default
+            }
+          }
+        } else {
+          // already published
+          this.buttons = {
+            primary: {
+              text: 'Save & Publish',
+              action: () => this.save(),
+              disabled: !this.isChanged,
+              // color: 'blue' // button default
+            }
+          }
+        }
+      }
+      break;
+    }
+
+    // secondary button if !isNew
+    if (this.id && this.buttons) {
+      if (this.currentStatus !== 'published') {
+        this.buttons.secondary = {
+          text: 'Delete',
+          action: (event) => this.confirmDelete(event),
+          color: 'red'
+        }
+      } else {
+        this.buttons.secondary = {
+          text: 'Unpublish',
+          action: (event) => this.unpublish(event),
+          color: 'red'
+        }
+      }
+    }
+  }
+
+  get working() {
+    return this.story &&
+      (this.story.isSaving || this.isPublishing ||
+        (this.story.versions && this.story.versions.some(version => version.files.some(file => file.isProcessing || file.isUploading))));
+  }
+
+  save() {
+    const wasNew = this.story.isNew;
+    this.story.save().subscribe(() => {
+      this.toastr.success('Episode saved');
+      if (wasNew) {
+        this.router.navigate(['/story', this.story.id]);
+      }
+    });
+  }
+
+  saveAndPublish(toast: string) {
+    const wasNew = this.story.isNew;
+    this.story.save().subscribe(() => {
+      // a story should not publish while audio is processing
+      if (this.story.versions.every(version => version.files.every(file => !file.isProcessing))) {
+        this.togglePublish(toast);
+      } else {
+        const poll = interval(2000).pipe(
+          mergeMap(() => this.story.doc.reload()),
```

Comment:
Seems like I could just reload the files and not the whole story here.

Reply:
The AudioFileModels are already polling for changes, right?  (That's how the processing-progressbar gets data).  Would it be possible to just look at them, instead of reloading the story?

That said ... this is also fine.

Reply:
you were absolutely correct on this, thank you! fixed